core/cmd/client.go

```diff
@@ -119,6 +120,15 @@ func AdvisoryLock(ctx context.Context, lggr logger.Logger, db *sql.DB, timeout t
 	for {
 		lockCtx, cancel := context.WithTimeout(ctx, timeout)
 		gotLock, err := lock.Lock(lockCtx)
+		if errors.Is(err, sql.ErrConnDone) {
+			lggr.Warnw("DB connection was unexpectedly closed; checking out a new one", "err", err)
+			// Initialize new lock (opening new connection) if the current connection is dead
+			if lock, err = pg.NewLock(initCtx, lockID, db); err != nil {
+				cancel()
+				return nil, err
+			}
+			gotLock, err = lock.Lock(lockCtx)
```

Comment:
Could it be a problem to reuse the same `lockCtx` here? What about calling `continue` instead to get a fresh timeout, plus eliminate the duplicate `lock.Lock` calls?
```suggestion
			continue
```

Reply:
Oh that changes err handling too though :shrug: